src/util/ajax.ts

```
@@ -159,7 +179,7 @@ function makeFetchRequest(requestParameters: RequestParameters, callback: Respon
                 return finishRequest(response, cacheableResponse, requestTime);
 
             } else {
-                return callback(new AJAXError(response.statusText, response.status, requestParameters.url));
+                return response.blob().then(body => callback(new AJAXError(response.status, response.statusText, requestParameters.url, body)));
```

This is not the same as returning a callback I think...?

---

Before the change, I don't think the `return` was required, as `callback` returns `void` and the result of the promise returned by `then` was never used. It is required now so that if calling `blob` or `callback` throws, the `catch` handler below handles the error.

---

As far as I know the callback should return a cancellable object to allow cancelation, which is not what a promise is, this is at least what I remember for this file...

---

I don't think I follow. `callback` has type `ResponseCallback<any>` which is a function that returns `void`. Isn't the cancellable object actually returned here: https://github.com/maplibre/maplibre-gl-js/blob/4cc4711c2a5fbd077f52db9b6b442a935bd3b0e5/src/util/ajax.ts#L201-L204

---

Right, this is a different path. Seems like the return value is not used.
I'm not sure what's the right approach here.
If the callback/promise value is never used I would prefer that the method inside the fetch will not return anything.
Simply call:
```
callback();
return;
```
Wouldn't it be more readable?

---

We still need to propagate any errors the `Response#blob` method might throw to the `catch` handler below.

---

Fair enough. If there will be issues here I'll tag you :-)